src/odemis/acq/stream/_static.py

```diff
@@ -389,22 +400,27 @@ def __init__(self, name, data, *args, **kwargs):
                         sempos_cur = sempos
                         break
                 self._pos[sempos_cur + (d.metadata.get(MD_POL_MODE, None),)] = img.ensure2DImage(d)
+
+                # if polarimetry info available, collect all pol pos data for each ebeam pos
+                if model.MD_POL_MODE in d.metadata.keys():
```

Comment:
There a couple of simplifications that can be done. I'd suggest to write something like:
```
pol = d.metadata.get(MD_POL_MODE, None)
pos_dict = self._pos_polarimetry.setdefault(sempos_cur, {})
pos_dict[pol] = img.ensure2DImage(d)
```